### PR TITLE
E2E Test: Vulnerability detector - Linux and Windows

### PR DESCRIPTION
**Pull request by BelenValdivia**
|Related issue|
|-------------|
|        https://github.com/wazuh/wazuh-qa/issues/3038     |

## Description

The test research was accomplished in https://github.com/wazuh/wazuh-qa/issues/2863. So, in this issue, we develop the E2E test to cover those use cases.

### Added

- Vulnerability detection tests:

1.  Ubuntu 20.04
2. Windows

### Configuration options

To launch these tests, we must indicate the path of the inventory:

```
python -m pytest -vvvs tests/end_to_end/test_vulnerability_detector/ --inventory_path=/home/belen/Repositories/wazuh-qa/tests/end_to_end/general_playbooks/inventory.yml 
```
It should be noted that, for local environments, it will be necessary to create the inventory for the moment. It should contain the following:

```
all:
  hosts:
    wazuh-manager:
      ansible_connection: ssh
      ansible_user: vagrant
      ansible_ssh_private_key_file: <path_to_private_key>
      ansible_python_interpreter: /usr/bin/python3
     
    wazuh-windows:
      ansible_user: vagrant
      ansible_password: vagrant
      ansible_connection: winrm
      ansible_winrm_server_cert_validation: ignore
      ansible_winrm_transport: basic
      ansible_winrm_port: 5985
      ansible_python_interpreter: C:\Users\vagrant\AppData\Local\Programs\Pyhton\Python39\python.exe
```
---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia (Developer)  |  test_vulnerability_detector/         | :no_entry_sign: :no_entry_sign: :no_entry_sign: |[ 🟢](https://github.com/wazuh/wazuh-qa/files/9155722/R1-tests-vulnerability-detector.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9155723/R2-tests-vulnerability-detector.zip)[🟢](https://github.com/wazuh/wazuh-qa/files/9155724/R3-tests-vulnerability-detector.zip) |     Ubuntu 20.04 and Windows server 2019    | be80f7d5fc36592d8b881ece76940ab1cc1c36e1     | Nothing to highlight |
| @juliamagan  (Reviewer)   |       test_vulnerability_detector/       | :no_entry_sign: :no_entry_sign: :no_entry_sign:|  [🟢](https://github.com/wazuh/wazuh-qa/files/9158752/R1-3090-test-vulnerability-detector.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9158756/R2-3090-test-vulnerability-detector.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9158760/R3-3090-test-vulnerability-detector.zip)  |   Ubuntu 20.04 and Windows server 2019      |    be80f7d5fc36592d8b881ece76940ab1cc1c36e1     | Nothing to highlight |

### CODE REVIEW

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ, but we need to implement just one test with two test cases instead of two tests.

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector_linux/data/test_cases/cases_vulnerability_detector.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+- name: vulnerability_detector
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector_linux/data/test_cases/cases_vulnerability_detector.yaml]: Done in 21e1225c653194a9a90fd6564e94b5dc03464250

File: tests/end_to_end/test_vulnerability_detector/data/playbooks/teardown_playbook.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+- name: Cleanup environment
```
Comment (juliamagan) [inline tests/end_to_end/test_vulnerability_detector/data/playbooks/teardown_playbook.yaml:1]: Please, rename the playbook to `teardown.yaml`. Also, we need to clean the `ossec.conf`.

File: tests/end_to_end/test_vulnerability_detector/data/test_cases/cases_vulnerability_detector.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+- name: detect_vulnerability
```
Comment (juliamagan) [inline tests/end_to_end/test_vulnerability_detector/data/test_cases/cases_vulnerability_detector.yaml:1]: We should implement two test cases, one for each os.

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ, but some changes are required. Please, update your branch and. Then, more changes will be needed in test_vulnerability_detector.py.

File: tests/end_to_end/test_vulnerability_detector/data/test_cases/cases_vulnerability_detector.yaml (hunk 3 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_vulnerability_detector/data/test_cases/cases_vulnerability_detector.yaml]: Done in 50b8bfccac4a8d86968c8312a76d406d9f104a98

File: tests/end_to_end/test_vulnerability_detector/data/playbooks/teardown_playbook.yaml (hunk 2 above)
Comment (BelenValdivia) [reply tests/end_to_end/test_vulnerability_detector/data/playbooks/teardown_playbook.yaml]: Done in 042087b6a5e452e63b51eba402508835b21dc6e3

**Review by juliamagan — APPROVED** [review]
LGTM

**Review by jmv74211 — APPROVED** [review]

File: tests/end_to_end/test_vulnerability_detector/data/playbooks/generate_events.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    - name: "{{ event_description }}"
+      ansible.windows.win_shell: "{{ command }}"
+      when: os == ansible_facts['os_family']
+
+- name: Wait alert
+  hosts: wazuh-manager
+  tasks:
+
+    - name: Wait for alert
```
[the remark this answers is not in the document]
Comment (BelenValdivia) [reply tests/end_to_end/test_vulnerability_detector/data/playbooks/generate_events.yaml]: Done in 1b4c4d88c3fb5114942f109e95daccdd907efcec

**Pull request merged**